src/helpers/vyos-event-handler.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2022 VyOS maintainers and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+import select
+import re
+from sys import argv
+
+from systemd import journal
+from vyos.util import call
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-p", "--pattern", action="store", help="Get pattern to match")
+parser.add_argument("-s", "--script", action="store", help="Get script to execute")
+
+data = journal.Reader()
+data.seek_tail()
+data.get_previous()
+p = select.poll()
+p.register(data, data.get_events())
+
+
+if __name__ == '__main__':
+    args = parser.parse_args()
+    if len(argv) != 5:
+        exit(1)
+    pattern = args.pattern
+    script = args.script
+
+    while p.poll():
+        if data.process() != journal.APPEND:
+            continue
+        for entry in data:
+            message = entry['MESSAGE']
+            if message != "":
+                result = re.search(rf'{pattern}', message)
```

Comment:
As @zdc suggests, we should pre-compile regex objects to avoid overhead from parsing them every time. That overhead isn't insignificant.